includes/helpers/class-accredible-learndash-admin-table-helper.php

```diff
@@ -168,6 +158,48 @@ private static function table_cell( $cell_value, $classes = '' ) {
 			return $start_cell_tag . $cell_value . '</td>';
 		}
 
+		/**
+		 * Evaluates Accredible group ID to string.
+		 *
+		 * @param int $group_id Accredible Group ID.
+		 *
+		 * @return string
+		 */
+		private static function eval_group_id( $group_id ) {
+			if ( null === self::$group_name_cache ) {
+				self::store_group_name_cache();
+			}
+
+			$key = $group_id;
+			if ( array_key_exists( $key, self::$group_name_cache ) ) {
+				$value = self::$group_name_cache[ $key ];
+			} else {
+				$value = 'Not found';
+			}
+
+			return $value;
+		}
+
+		/**
+		 * Store group_name_cache.
+		 */
+		private static function store_group_name_cache() {
+			self::$group_name_cache = array();
+			$page                   = 1;
+			$client                 = new Accredible_Learndash_Api_V1_Client();
+			while ( ! empty( $page ) ) {
+				$response = $client->search_groups( null, $page, 50 );
+				if ( isset( $response['errors'] ) ) {
+					wp_die( 'Accredible API Search Groups Error ' . esc_attr( $response['errors'] ) );
+				}
+
+				foreach ( $response['groups'] as $group ) {
+					self::$group_name_cache[ $group['id'] ] = $group['name'];
+				}
+				$page = $response['meta']['next_page'];
```

Comment:
The current group search endpoint does not return `meta` but we are aware of that and we've already fixed this behavior by NTGR-597. The change will be released soon. Until then, `$response['meta']['next_page']` is always `null`, but I expect most of our clients have less than 50 groups.